pkg/processor/build/runtime/dotnetcore/runtime.go

```diff
@@ -0,0 +1,196 @@
+	// in the staging directory
+	if err := d.buildHandler(stagingDir); err != nil {
+		return errors.Wrap(err, "Failed to build handler plugin")
+	}
+
+	return nil
+}
+
+// GetName returns the name of the runtime, including version if applicable
+func (d *dotnetcore) GetName() string {
+	return "dotnetcore"
+}
+
+func (d *dotnetcore) createUserFunctionPath(stagingDir string) error {
+	userFunctionPathInStaging := filepath.Join(stagingDir, "/function")
+	d.Logger.DebugWith("Creating user function path", "path", userFunctionPathInStaging)
+
+	if err := os.MkdirAll(userFunctionPathInStaging, 0755); err != nil {
+		return errors.Wrapf(err, "Failed to create user function path in staging at %s", userFunctionPathInStaging)
+	}
+
+	copyFrom := d.GetFunctionDir()
+	d.Logger.DebugWith("Copying user function", "from", copyFrom, "to", userFunctionPathInStaging)
+
+	_, err := util.CopyDir(copyFrom, userFunctionPathInStaging)
+	if err != nil {
+		return errors.Wrapf(err, "Error copying from %s to %s", copyFrom, userFunctionPathInStaging)
+	}
+
+	return nil
+}
+
+func (d *dotnetcore) buildHandler(stagingDir string) error {
+
+	// build the image that builds the handler. it will contain the handler when it's done
+	// and/or a handler_build.log
+	if err := d.buildHandlerBuilderImage(stagingDir); err != nil {
+		return errors.Wrap(err, "Failed to build handler builder image")
+	}
+
+	// delete the image when we're done
+	defer d.DockerClient.RemoveImage(handlerBuilderImageName)
+
+	// the staging paths of the files we want to copy
+	handlerBinaryPathInStaging := path.Join(stagingDir, "handler")
+	sdkPathInStaging := path.Join(stagingDir, "nuclio-sdk-dotnetcore")
+	wrapperBinaryPathInStaging := path.Join(stagingDir, "wrapper")
+	processorBinaryPathInStaging := path.Join(stagingDir, "processor")
+
+	// copy artifacts from the image we build - these directories are defined
+	// in the onbuild dockerfile. we allow copy errors because processor may not
+	// exist
+	objectsToCopy := map[string]string{
+		"/usr/local/bin/processor":          processorBinaryPathInStaging,
+		"/opt/nuclio/handler":               handlerBinaryPathInStaging,
+		"/opt/nuclio/wrapper":               wrapperBinaryPathInStaging,
+		"/opt/nuclio/nuclio-sdk-dotnetcore": sdkPathInStaging,
+	}
+
+	if err := d.DockerClient.CopyObjectsFromImage(handlerBuilderImageName, objectsToCopy, false); err != nil {
+		return errors.Wrap(err, "Failed to copy objects from image")
+	}
+
+	d.Logger.DebugWith("Successfully built and copied handler plugin", "path", handlerBinaryPathInStaging)
+
+	return nil
+}
+
+func (d *dotnetcore) buildHandlerBuilderImage(stagingDir string) error {
+
+	versionInfo, err := version.Get()
+	if err != nil {
+		return errors.Wrap(err, "Failed to get version info")
+	}
+
+	handlerBuilderOnBuildImageName := fmt.Sprintf("nuclio/handler-builder-dotnetcore-onbuild:%s-%s",
+		versionInfo.Label,
+		versionInfo.Arch)
+	d.Logger.Info("Building from %s", handlerBuilderOnBuildImageName)
```

Comment:
Please use structured logging (works much better with appinsights :) and also degrade to "Debug" as this will needlessly appear in non-verbose logs:

`d.Logger.DebugWith("Building", "from", handlerBuilderOnBuildImageName)`

(same for other places).